woocommerce/google-listings-and-ads
# Simplify add and update shipping rates into upsert shipping rates.

### Changes proposed in this Pull Request:

Closes #126 . This is a follow up to https://github.com/woocommerce/google-listings-and-ads/pull/138#discussion_r571198592 (special thanks to @tomalec)

This PR simplifies the call to add and update shipping rates into upsert shipping rates, which are both done using `HTTP POST`.

In addition, it also fixes a current bug where if you have an existing rate (e.g. "China") and you try to add a new rate with "China" again, you see will two "China"s in the frontend. The two "China"s is stored in the redux store. If you refresh the page, there will be only one "China".

### Detailed test instructions:

1. Open https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc
2. Proceed to Step 3. 
3. Add, edit and delete the shipping rates and make sure they are saved correctly.
    - Edit includes clicking on the edit button to open the edit rate modal, and also editing inline in the input box without clicking on the edit button.

### Changelog Note:

- Enhancement: simplify add and update shipping rates.
- Fix: bug with duplicated shipping rate countries.
